workflows/sampletracking.nf

```diff
@@ -40,12 +43,78 @@ workflow SAMPLETRACKING {
     def ch_multiqc_files = Channel.empty()
     def ch_pool_multiqc_files = Channel.empty()
 
+
+    def (ch_sample, ch_snp, ch_rest) = ch_samplesheet
+        .multiMap {meta, sample_bam, sample_bam_index, snp_fastq, snp_bam, snp_bam_index ->
+            sample: [meta, (sample_bam instanceof List && sample_bam.isEmpty() ? null : sample_bam), (sample_bam_index ?: null)]
+            snp   : [meta, (snp_bam   instanceof List && snp_bam.isEmpty()   ? null : snp_bam),     (snp_bam_index   ?: null)]
+            rest  : [meta, (snp_fastq instanceof List && snp_fastq.isEmpty() ? null : snp_fastq)]
+        }
+
+    def (ch_sample_with_idx, ch_sample_no_idx) = ch_sample.branch {
+        _meta, _sample_bam, sample_bam_index ->
+            with_index: sample_bam_index != null
+            no_index  : sample_bam_index == null
```

Comment:
```suggestion
            with_index: sample_bam_index
            no_index  : !sample_bam_index
```
`[]` and `null` are considered a `false` value in Groovy. Having it like this makes your code more flexible, which is wanted in this case